src/tests/test_bbox.c

```diff
@@ -0,0 +1,145 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifdef _WIN32
+# include <io.h>
+# define unlink _unlink
+#else
+# include <unistd.h>
+#endif
+#include "cgnslib.h"
+
+#ifndef CGNSTYPES_H
+# define cgsize_t int
+#endif
+#ifndef CGNS_ENUMT
+# define CGNS_ENUMT(e) e
+# define CGNS_ENUMV(e) e
+#endif
+
+#define NUM_SIDE 5
+
+float coord[NUM_SIDE*NUM_SIDE*NUM_SIDE];
+
+#define CHECK(L,B) if(!(B)){fprintf(stderr,"mismatch in %s\n",L);exit(1);}
+
+void error_exit (char *where)
+{
+    fprintf (stderr, "ERROR:%s:%s\n", where, cg_get_error());
+    exit (1);
+}
+
+int main (int argc, char **argv)
+{
+    int n, i;
+    int cgfile, cgbase, cgzone, cgcoord;
+    cgsize_t size[9];
+    static char *fname = "boundingbox.cgns";
+    double bbox[2][3];
+
+
+    if (argc > 1) {
+        n = 0;
+        if (argv[1][n] == '-') n++;
+        if (argv[1][n] == 'a' || argv[1][n] == 'A')
+            i = CG_FILE_ADF;
+        else if (argv[1][n] == 'h' || argv[1][n] == 'H')
+            i = CG_FILE_HDF5;
+        else {
+            fprintf(stderr, "unknown option\n");
+            exit (1);
+        }
+        if (cg_set_file_type(i))
+            error_exit("cg_set_file_type");
+    }
+
+    for (n = 0; n < NUM_SIDE*NUM_SIDE*NUM_SIDE; n++)
+        coord[n] = (float)n;
+
+    unlink (fname);
+    printf ("creating CGNS file %s\n", fname);
+    if (cg_open (fname, CG_MODE_WRITE, &cgfile) ||
+        cg_base_write (cgfile, "Base", 3, 3, &cgbase))
+        error_exit ("write base");
+
+
+    /* write zone */
+
+    puts("writing zone");
+    for (n = 0; n < 3; n++) {
+        size[n]   = NUM_SIDE;
+        size[n+3] = NUM_SIDE - 1;
+        size[n+6] = 0;
+    }
+    if (cg_zone_write (cgfile, cgbase, "Zone", size, CGNS_ENUMV(Structured), &cgzone) ||
+        cg_coord_write(cgfile, cgbase, cgzone, CGNS_ENUMV(RealSingle),
+            "CoordinateX", coord, &cgcoord) ||
+        cg_coord_write(cgfile, cgbase, cgzone, CGNS_ENUMV(RealSingle),
+            "CoordinateY", coord, &cgcoord) ||
+        cg_coord_write(cgfile, cgbase, cgzone, CGNS_ENUMV(RealSingle),
+            "CoordinateZ", coord, &cgcoord))
+        cg_error_exit();
+
+    puts ("closing and reopening in read mode");
+    cg_close (cgfile);
```

Comment:
all cg API's should check for an error in the tests.

Reply:
done